raft.go

```diff
@@ -1062,7 +1062,7 @@ func (r *raft) Step(m pb.Message) error {
 	case m.Term > r.Term:
 		if m.Type == pb.MsgVote || m.Type == pb.MsgPreVote {
 			force := bytes.Equal(m.Context, []byte(campaignTransfer))
-			inLease := r.checkQuorum && r.lead != None && r.electionElapsed < r.electionTimeout
+			inLease := r.checkQuorum && r.lead != None && r.electionElapsed < r.electionTimeout && r.lead != m.From
```

Comment:
Did you see a real case that a leader sent vote or pre-vote (`r.lead == m.From`) and when the lease hasn't expired yet (`r.electionElapsed < r.electionTimeout`)?

It will never happen. Please create a case to reproduce it if you still think it's an issue.

Reply:
FYI. 

https://github.com/etcd-io/raft/blob/2f109970c68e505c223f4d3d5c13a8dcd8f7d528/raft.go#L948-L951

Reply:
Sorry, I mistakenly thought r.electionTimeout is random, but r.randomizedElectionTimeout really is,